Core/src/org/sleuthkit/autopsy/geolocation/GeolocationTopComponent.java

```diff
@@ -0,0 +1,250 @@
+        });
+    }
+    
+    @Override
+    public void addNotify() {
+        super.addNotify();
+        IngestManager.getInstance().addIngestModuleEventListener(INGEST_MODULE_EVENTS_OF_INTEREST, ingestListener);
+        Case.addEventTypeSubscriber(EnumSet.of(CURRENT_CASE), evt -> {
+            mapPanel.clearWaypoints();
+            if(evt.getNewValue() != null) {
+                initWaypoints();
+            }
+        });
+    }
+
+    @Override
+    public void removeNotify() {
+        super.removeNotify();
+        IngestManager.getInstance().removeIngestModuleEventListener(ingestListener);
+    }
+    
+    /**
+     * Set the state of the refresh panel at the top of the mapPanel.
+     * 
+     * @param show Whether to show or hide the panel.
+     */
+    private void showRefreshPanel(boolean show) {
+        if(show) {
+            mapPanel.add(refreshPanel, BorderLayout.NORTH);
+        } else {
+            mapPanel.remove(refreshPanel);
+        }
+        mapPanel.revalidate();
+    }
+
+    /**
+     * Use a SwingWorker thread to find all of the artifacts that have GPS
+     * coordinates.
+     *
+     */
+    private void initWaypoints() {
+        SwingWorker<List<Waypoint>, Waypoint> worker = new SwingWorker<List<Waypoint>, Waypoint>() {
+            @Override
+            protected List<Waypoint> doInBackground() throws Exception {
+                List<Waypoint> waypoints = new ArrayList<>();
+
+                Case currentCase;
+                try {
+                    currentCase = Case.getCurrentCaseThrows();
+                } catch (NoCurrentCaseException ex) {
+                    logger.log(Level.WARNING, "Unable to get artifacts for geolocation window, no current open", ex);
+                    // Popup a message or something?
+                    return waypoints;
+                }
+
+                // TSK_GPS_TRACKPOINT, TSK_GPS_SEARCH, TSK_GPS_LAST_KNOWN_LOCATION 
+                // and TSK_GPS_BOOKMARK have similar attributes and can be processed
+                // similarly
+                List<BlackboardArtifact> artifacts = new ArrayList<>();
+                artifacts.addAll(currentCase.getSleuthkitCase().getBlackboardArtifacts(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_TRACKPOINT));
+                artifacts.addAll(currentCase.getSleuthkitCase().getBlackboardArtifacts(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_SEARCH));
+                artifacts.addAll(currentCase.getSleuthkitCase().getBlackboardArtifacts(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_LAST_KNOWN_LOCATION));
+                artifacts.addAll(currentCase.getSleuthkitCase().getBlackboardArtifacts(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_BOOKMARK));
+                
+                for (BlackboardArtifact artifact : artifacts) {
+                    Waypoint point = new SimpleArtifactWaypoint(artifact);
+
+                    if (point.getPosition() != null) {
+                        waypoints.add(new SimpleArtifactWaypoint(artifact));
```

Comment:
we cant do waypoints.add(point) here. 

no need to create same artifact again?